src/tools/wasm2wast.cc

```diff
@@ -156,9 +156,13 @@ int main(int argc, char** argv) {
       }
 
       if (WABT_SUCCEEDED(result)) {
-        std::unique_ptr<FileWriter> writer(s_outfile ? new FileWriter(s_outfile)
-                                                     : new FileWriter(stdout));
-        result = write_ast(writer.get(), &module);
+        if (s_outfile) {
+          FileWriter writer(s_outfile);
+          result = write_ast(&writer, &module);
+        } else {
+          FileWriter writer(stdout);
+          result = write_ast(&writer, &module);
```

Comment:
That's what I meant before, it's not allowed because they are different constructors. The ternary operator happens first which wants to coerce `const char*` and `FILE*` to some common type but can't.